django/db/models/base.py

```diff
@@ -1310,7 +1310,7 @@ def _get_field_expression_map(self, meta, exclude=None):
         meta = meta or self._meta
         field_map = {}
         generated_fields = []
-        for field in meta.local_concrete_fields:
+        for field in meta.local_fields:
```

Comment:
I think this introduces a performance regression: both `patron` and `patron_id` will be checked against the db even though they're backed by the same column. There might be a different solution making use of `field.local_related_fields`. 🤔